drivers/ethernet/nxp_enet/eth_nxp_enet.c

```diff
@@ -566,18 +572,48 @@ static void eth_nxp_enet_isr(const struct device *dev)
 	irq_unlock(irq_lock_key);
 }
 
+/* Note this is not universally unique, it just is probably unique on a network */
 static inline void nxp_enet_unique_mac(uint8_t *mac_addr)
 {
 	uint32_t id = ETH_NXP_ENET_UNIQUE_ID;
 
-	mac_addr[0] = FREESCALE_OUI_B0;
+	/* Setting LAA bit because it is not guaranteed universally unique */
+	mac_addr[0] = FREESCALE_OUI_B0 | 0x02;
 	mac_addr[1] = FREESCALE_OUI_B1;
 	mac_addr[2] = FREESCALE_OUI_B2;
 	mac_addr[3] = FIELD_GET(0xFF0000, id);
 	mac_addr[4] = FIELD_GET(0x00FF00, id);
 	mac_addr[5] = FIELD_GET(0x0000FF, id);
 }
 
+#ifdef CONFIG_SOC_FAMILY_NXP_IMXRT
+#include <fsl_ocotp.h>
+#endif
+
+static inline void nxp_enet_fused_mac(uint8_t *mac_addr)
```

Comment:
Can you return `-ENOTSUP` for unsupported SoCs, and pass that to upper layers?

Reply:
maybe instead of this I will add a build assert